NOTES.md

```diff
@@ -0,0 +1,35 @@
+# Outline
+
+## Middleware
+
+- Fetches EUR conversion hourly/yearly -- resets cache
+- API:
+  - '/api/currencies/:currencyCode' -- returns currency object with all conversions, adds result to cache
+  - additionally, every return should include a marker indicating how long it will be valid for
+
+## Extension
+
+- On user request, converts the displayed currencies to the preferred base currency
+
+### Content script
+
+- Responsible for parsing the currencies on the page and reporting to the background script
```

Comment:
Could report classes/IDs of elements that contained a currency conversion to be cached in storage so the DOM doesn't have to be searched each time for similar pages within the same (sub)domain/page

Reply:
Or we could cache the path of DOM nodes that lead to conversions, for example:

```
<body>
  <div>
    <div class="flight"><span>$</span>99</div>
    <div class="flight"><span>$</span>145</div>
  </div>
</body
```
This would get cached as `[0, [0, [0,1]]]` for `root -> first child -> first child, second child`

Reply:
I wonder which would be easier, caching the selector or the node path